docs/ml-process.md

```diff
@@ -0,0 +1,318 @@
+The task that is running this script [can be found here](https://github.com/mozilla/bugbug/blob/3e4f452b5e523d3d114cb0477e5a82f22c7e951e/infra/data-pipeline.yml#L232-L270) 
+and results of the task are [located here](https://community-tc.services.mozilla.com/tasks/index/project.bugbug.data_github_issues/latest).
+
+If you run the above command locally, it will look for the existing dataset and check its `last_modified` timestamp (the dataset is saved as an [artifact of the task](https://community-tc.services.mozilla.com/api/index/v1/task/project.bugbug.data_github_issues.latest/artifacts/public%2Fgithub_issues.json.zst) in the Taskcluster). 
+If the dataset is available, it will be downloaded to `bugbug/data` folder to your machine and the script will begin fetching updates from GitHub and save them to this local copy. 
+
+During this update, all changes to issues since `last_modified` timestamp will be fetched. If the dataset or its `last_modified` is not available, the script will begin fetching all issues starting from issue 1.
+
+
+## 2. Training the model
+We use [needsiagnosis model](https://github.com/mozilla/bugbug/blob/master/bugbug/models/needsdiagnosis.py) for classifying incoming issues. 
+
+The goal of the model is to find issues that don't need to be diagnosed. We look through lifecycle events of each issue and if an issue
+has ever been moved to `needsdiagnosis` milestone, we assign a `label 0` (needsdiagnosis=true), and if not `label 1`(needsdiagnosis=false):
+
+```python
+         classes = {}
+
+
+        for issue in github.get_issues():
+            # Skip issues that are not moderated yet as they don't have a meaningful title or body
+            if issue["title"] == "In the moderation queue.":
+                continue
+
+
+            for event in issue["events"]:
+                if (
+                    event["event"] == "milestoned"
+                    and event["milestone"]["title"] == "needsdiagnosis"
+                ):
+                    classes[issue["number"]] = 0
+
+
+            if issue["number"] not in classes:
+                classes[issue["number"]] = 1
+
+
+        logger.info(
+            f"{sum(1 for label in classes.values() if label == 1)} issues have not been moved to needsdiagnosis"
+        )
+        logger.info(
+            f"{sum(1 for label in classes.values() if label == 0)} issues have been moved to needsdiagnosis"
+        )
+
+
+        return classes, [0, 1]
+```
+
+We're using issue `title` and `body` as our training data:
+
+```python
+                    ColumnTransformer(
+                        [
+                            ("title", self.text_vectorizer(min_df=0.0001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.0001),
+                                "first_comment",
+                            ),
+                        ]
+                    ),
+```
+
```

Comment:
what does `min_df` mean?

Reply:
I've added an explanation of this parameter too